### DISCUSSION TITLE
Building new packages

### DISCUSSION THREAD

**Discussion by fullmetal-fred** [kairos-io/kairos#529] [Q&A]
Hey there! Is there a good documentation source for building bundles/packages for use with Kairos? 

Use case is: I'm using docker for my container runtime. Right now I'm installing it via commands in the cloud-config. But I have a feeling the preferred method would be to define a bundle and just add the bundle. Thoughts? Am I on the right track? I also have a variation that is a whole Dockerfile based on Kairos core...but I'd like to not have to maintain it if I can help it. 

Thanks!

**Comment by mudler** [comment]
Hi @cybernetic-mountaineer, that's a good question!

It's unfortunate that currently docs are not right there yet for bundles. Bundles will officially go supported in https://github.com/kairos-io/kairos/issues/116 which should allow then to consume bundles easily. 

In the meantime - there are 2 ways to add additional packages to Kairos without [customizing the container image](https://kairos.io/docs/advanced/customizing/):
- Create bundles
  - those are just naked container images containing the files you want to add - but they are meant to install everything in `/usr/local` which is mapped to `COS_PERSISTENT`. The installation of bundles is done during runtime - so everything is immutable except for `/usr/local`
    - alternatively they can also install things outside `/usr/local` - but then those can be consumed with `osbuilder` and/or directly with Dockerfiles
  - Bundles can be also "executed" - that means you can carefully create a container image that acts as "installer" for what you want to install in the system. Consider it will be executed in runtime, so only `/usr/local` is persistent.
- Use [Live layering](https://kairos.io/docs/advanced/livelayering/) - this allows to add system overlays on top of `/usr` - there is an example in the page which more or less can be used for bundles too. The only difference is where the files are installed, and how it is consumed

**Reply by liyimeng** [reply]
what about luet package? if I, e.g. wanna add a k3s version is not available, how do I create a luet package? would be nice to add some concrete helping text at https://github.com/kairos-io/packages.
